float64 new_r
int32 new_N
---
float64 old_r
int32 old_N